docs/targets/solana.rst

```diff
@@ -1,34 +1,100 @@
+++++++++++
+
+- An account address consists of a 32-bytes key, which is represented by the ``address`` type. This data model
+  differs from Ethereum 20-bytes addresses.
+- Solana's virtual machine registers are 64-bit wide, so 64-bit integers ``uint64`` and ``int64`` are preferable over
+  ``uint256``, ``int256``, ``uint128``, or ``int128``. An operation with types wider than 64-bits is split into multiple instructions, making
+  it slower and consuming more compute units. Multiplication, division and modulo will require even more instructions,
+  depending what values are involved. In addition, 256 bit values require more storage, either in account data, on the
+  stack, or on the heap.
+- Likewise, all balances and values on Solana are 64-bit wide, so the builtin functions for
+  *address* ``.balance``, ``.transfer()`` and ``.send()`` use 64-bit integers.
+- An address literal has to be specified using the ``address"36VtvSbE6jVGGQytYWSaDPG7uZphaxEjpJHUUpuUbq4D"`` syntax.
+- Ethereum syntax for addresses ``0xE0f5206BBD039e7b0592d8918820024e2a7437b9`` is not supported.
+
+Runtime
++++++++
+
+- The Solana target requires `Solana <https://www.solana.com/>`_ v1.8.1.
+- Function selectors are eight bytes wide and known as *discriminators*.
 - Solana provides different builtins, e.g. ``tx.program_id`` and ``tx.accounts``.
-- Function selectors are eight bytes and known as *discriminators*.
 
-This is how to build your Solidity for Solana:
+Compute budget
+++++++++++++++
+
+On Ethereum, when calling a smart contract function, one needs to specify the amount of gas the operation is allowed
+to use. Gas serves to pay for a contract execution on chain and can be a way for giving a contract priority execution
+when extra gas is offered in a transaction. Each EVM instruction has an associated gas value, which translates to real
+ETH cost. Provided that one can afford all the gas expenses, there is no upper boundary for the amount of gas limit
+one can provide in a transaction, so Solidity for Ethereum has gas builtins, like ``gasleft``, ``block.gaslimit``,
+``tx.gasprice`` or the Yul ``gas()`` builtin, which returns the amount of gas left for execution.
+
+On the other hand, Solana is optimized for low latency and high transaction throughput and has an equivalent concept to
+gas: compute unit. Every smart contract function is allowed the same quantity of compute units (currently that
+value is 200k), and every instruction of a contract consumes exactly one compute unit. There is no need to provide
+an amount of compute units for a transaction and they are not charged, except when one wants priority execution on
+chain, in which case one would pay per compute unit consumed. Therefore, functions for gas are not available on
+Solidity for Solana.
+
+
+Solidity for Solana incompatibilities with Solidity for Ethereum
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+- There is no ``msg.sender``, as detailed :ref:`here <msg_sender_solana>`.
```

Comment:
making the text of a link `here` or `this` is bad style but this PR is taking too long